### ISSUE TITLE
[Bug] Error when trying to use extension.writeStore() in extension

### ISSUE THREAD

**Issue by Blazeflack** [GilbertGobbels/GAwesomeBot#4]
When attempting to use extension.writeStore() you will be presented with an error.
Example code:

```
extension.writeStore("Test", "Something");
```

Error from Logs:

```
Failed to run command extension 'Project Release Announcement':
SyntaxError: Unexpected token u in JSON at position 0 at Object.parse (native)
at Extension.writeStore (/home/Gilbert/AwesomeBot/master/Modules/ExtensionStructures/Extension.js:13:28)
at evalmachine.<anonymous>:1:11
at Domain.extensionDomain.run (/home/Gilbert/AwesomeBot/master/Modules/ExtensionRunner.js:19:18)
at Domain.run (domain.js:221:14)
at fs.readFile.err (/home/Gilbert/AwesomeBot/master/Modules/ExtensionRunner.js:17:21)
at tryToString (fs.js:449:3) at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12) 
```

**Comment by GilbertGobbels** [comment]
Will try to recreate today

**Comment by ItsLachy** [comment]
Bump...

**Comment by HilbertGilbertson** [comment]
I've played around with this briefly and it seems extensionDocument.store is undefined, so the error results from trying to stringify and parse JSON from an array that doesn't exist.

I tried removing lines 13-17 & 27 in Extension.js, so the content of the else statement on line 17 is all that would run when triggering extension.writeStore(), and the result was absolutely nothing. No errors returned but it still wasn't saving.

**Comment by HilbertGilbertson** [comment]
I've provided a fix in #55, also resolving the same issue with deleteStore. As noted in the PR, there's still no extension.store() function but that's a very easy job if someone wants to code and submit it (if not I'll get round to it myself at some point later this week).

**Comment by Blazeflack** [comment]
Good job.
